content/en/blog/2023/native-sidecars/index.md

```diff
@@ -0,0 +1,207 @@
+While containers within a Pod can share some things, their *lifecycle's* are entirely decoupled.
+To Kubernetes, both of these containers are functionally the same.
+
+However, in Istio they are not the same - the Istio container is required for the primary application container to run,
+and has no value without the primary application container.
+
+This mismatch in expectation leads to a variety of issues:
+* If the application container starts faster than Istio's container, it cannot access the network.
+  This wins the [most +1's](https://github.com/istio/istio/issues/11130) on GitHub by a landslide.
+* If Istio's container shuts down before the application container, it cannot access the network.
+  This often means we shutdown too fast (breaking the network) or too slow (leaving pods running with no purpose).
+* If an application container intentionally exits (typically from usage in a `Job`), Istio's container will still run and keep the pod running indefinitely.
+  This is also a [top GitHub issue](https://github.com/istio/istio/issues/11659).
+* `InitContainers`, which run before Istio's container starts, cannot access the network.
+
+Countless hours have been spent in the Istio community and beyond to work around these - to limited success.
+
+## Fixing the root cause
+
+While increasingly-complex workarounds in Istio can help alleviate the pain for Istio users, ideally all of this would just work - and not just for Istio.
+Fortunately, the Kubernetes community has been hard at work to address these directly in Kubernetes.
+
+In Kubernetes 1.28, a new feature to add native support for sidecars was merged, closing out over 5 years of ongoing work.
+With this merged, all of our issues can be addressed without workarounds!
+
+While we are on the "GitHub issue hall of fame", [these](https://github.com/kubernetes/kubernetes/issues/25908) [issues](https://github.com/kubernetes/kubernetes/issues/65502) account for #1 and #6 all time issues in Kubernetes - and have finally been closed!
+
+A special thanks goes to the huge group of individuals involved in getting this past the finish line.
+
+## Trying it out
+
+While Kubernetes 1.28 is not out yet, and the feature is Alpha (and therefore, off by default), and the support for the feature in Istio is not yet shipped, we can still try it out today - just don't try this in prod!
+
+First, we need to spin up a pre-release Kubernetes cluster, with the `SidecarContainers` feature enabled:
+
+{{< text shell >}}
+$ cat <<EOF | kind create cluster --name sidecars --image gcr.io/istio-testing/kind-node:v1.28.0-beta.0 --config=-
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+featureGates:
+  SidecarContainers: true
+EOF
+{{< /text >}}
+
+Then we can download the latest Istio pre-release. I used Linux here; see [Dev Builds](https://github.com/istio/istio/wiki/dev-builds) for more steps.
+These images are built from the latest Istio commit, so again - do not try in prod!
```

Comment:
```suggestion
These images are built from the latest Istio commit, so again - do not try this in prod!
```

Reply:
Maybe we want to expand prod -> production here and in the comment above. (and elsewhere).